### PR TITLE
{WIP}(MODULES-2957) Allow backslash in registry value names

### PR DESCRIPTION
**Pull request by ferventcoder**

### CODE REVIEW

File: manifests/value.pp [hunk 1]
```diff
@@ -70,8 +68,9 @@
   # If value_real is an empty string then the default value of the key will be
   # managed.
   registry_value { "${key}\\${value_real}":
+    path => $key,
+    value_name => $value_real,
     type => $type,
     data => $data,
   }
 }
-
```
Comment (ferventcoder) [inline manifests/value.pp]: Likely causing the travis failure (the trailing line should be there...IDE editors :facepunch:)

File: lib/puppet/type/registry_value.rb [hunk 2]
```diff
@@ -50,6 +50,17 @@ def self.title_patterns
...
   end
 
+  # todo: if value_name is not set - during the right time - we will take the last value off of path and set it
+  newproperty(:value_name) do
+    desc "The name of the registry value to manage.  For example:
+      'Value1'. This is typically specified separately when the value
+      name includes backslashes. "
+
+    validate do |path|
+      # what validation should occur here
```
Comment (ferventcoder) [inline lib/puppet/type/registry_value.rb]: No validation - otherwise it won't be compatible.

Comment (ferventcoder) [inline lib/puppet/type/registry_value.rb]: We could provide a deprecation warning here though.

**Comment by glennsarti** [comment]
This is a duplicate of #101

**Comment by ferventcoder** [comment]
Actually #101 is a duplicate of this.

**Comment by Iristyle** [comment]
This is being superseded in #143 and #142 - as such, closing.

Thanks for the PR @ferventcoder!

**Comment by Iristyle** [comment]
#142 ended up introducing some unwanted failure modes - that effort has been closed, and reopened under #146

**Pull request closed without merging**